### PR TITLE
Re-introduce pytest-socket

### LINKED ISSUE #342: Re-investigate using pytest-socket
## Description

`pytest-socket` was removed in #276 because it was causing some inconsistencies with test completion when tests were run out of order. As discussed in #340, it might make sense to bring the package back to ensure that provider tests aren't making any unnecessary request calls during testing.

## Benefit

Where possible, all external requests in tests should be mocked so that we don't have to rely on the uptime/availability of the service in order to run our tests.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #342 by @AetherUnbound

## Description

This PR re-introduces pytest-socket, which ensures that tests don't attempt to reach out to services via a socket unless explicitly allowed. 

In attempting to uncover which tests were failing with pytest-socket, I found that the license URL validation [makes a request to Creative Commons to ensure the URL is accurate](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/urls.py#L65). This was the cause of almost all of our external accessing and was used internally by many tests across the codebase. The only way to mock it in its current form is to auto-use a global fixture that mocks all `request.get` calls, which has undesirable side effects for other tests. So I opted to alias `requests.get` within that module specifically, which made the mock much easier to patch (even though the patch still needed to remain as a global fixture).

A few tests required specific care:
* The Smithsonian unit code test was making a request to the Smithsonian API that was immediately return with an invalid API key error :joy: Since we don't really care what the response is, I mocked this one.
* The single run external DAG sensor test was emitting a warning that had nothing to do with any code we had written and everything to do with Airflow internals. I suppressed that warning for now so we can have cleaner test output.

In addition to making sure our tests don't rely on any external services, a fun side-effect of this is that our tests are now slightly faster!

## Testing Instructions

You'll need to rebuild the container in order to test with this new dependency:
1. `just build`
1. `just test`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
I'm working on addressing the CI failures. It turns out that:
* Docker will use different internal ports depending on what's available on the machine
* Boto3 will open a socket with the IP address directly rather than the hostname, for some reason
* Trying to get the host from the hostname dynamically isn't working with our S3 bucket fixture

**Comment by AetherUnbound** [comment]
Ah, okay, here's why: https://github.com/pytest-dev/pytest/issues/1368 :upside_down_face:

File: tests/dags/common/loader/test_smithsonian_unit_codes.py [hunk 1]
```diff
@@ -52,10 +52,14 @@ def test_alert_new_unit_codes():
...
     unit_code_table = SI_UNIT_CODE_TABLE
+    response_mock = Mock()
+    response_mock.json.return_value = {}
 
     with patch.object(
         si, "get_new_and_outdated_unit_codes", return_value=({"d"}, {"e"})
-    ) as mock_get_unit_codes:
+    ) as mock_get_unit_codes, patch.object(
+        si.requests, "get", return_value=response_mock
+    ):
```
Comment (stacimc) [inline tests/dags/common/loader/test_smithsonian_unit_codes.py:62]: FWIW this code will be deleted in #452, but I am mocking it in the [new tests](https://github.com/WordPress/openverse-catalog/pull/452/files#diff-b25f876380bf67a37d45141dba9bf98b9e88ba7194b7e2e2e02d5678172b4840R73) so it should still be good 😄

Comment (AetherUnbound) [reply tests/dags/common/loader/test_smithsonian_unit_codes.py:62]: NICE :fire:

**Review by stacimc — APPROVED** [review]
LGTM! Took me a minute  to reason through the changes, but I think the import aliasing is a good solution for the license url validation. The additional docs are always appreciated 😃

**Review by sarayourfriend — APPROVED** [review]
LGTM!

**Pull request merged**